### PR TITLE
Add error messages to MAE shell scripts

### PR DESCRIPTION
**Pull request by mumichae**
Whenever the output of GATK ASECounter is empty, DROP will continue with the empty files until DESeq2 throws an error.

One option is to throw an error every time the ASECounter output is empty. This is implemented here. There might be other possibilities in addressing this issue, so any discussion is welcome.

### CODE REVIEW

File: drop/modules/mae-pipeline/MAE/ASEReadCounter.sh [hunk 1]
```diff
@@ -57,10 +56,19 @@ for chr in $chr_subset; do
...
   -F $'\t' 'BEGIN {OFS = FS} NR==1{print $0, "ID"} NR>1{print $0, id}' |
   bgzip >${output}
 rm ${tmp}
 
 zcat ${output} | head
+
+num_out=$(zcat "${output}" | wc -l )
+if [ "${num_out}" -lt 2 ]
+then
+  echo 'ERROR: No allele-specific counts'
```
Comment (c-mertes) [inline drop/modules/mae-pipeline/MAE/ASEReadCounter.sh]: Please add a suggestion how to solve it. 
1. Checking the chromosomes (bam/vcf)
2. Check the reference -> having 1 reference/fasta file per BAM could also solve dome of the issues here
3. Removing the sample from MAE
4. ...

File: drop/modules/mae-pipeline/MAE/filterSNVs.sh [hunk 2]
```diff
@@ -51,5 +51,15 @@ else  # VCF and BAM have same chromosome format
...
 fi
 
+num_out=$(zcat "${output}" | grep -vc '#' )
+if [ "${num_out}" -eq 0 ]
+then
+  echo "ERROR: no entries after filtering for SNVs"
+  echo "VCF ID: ${vcf_id}"
+  echo "VCF file: ${vcf_file}"
+  echo "BAM file: ${bam_file}"
+  exit 1
```
Comment (c-mertes) [inline drop/modules/mae-pipeline/MAE/filterSNVs.sh]: Same as above. Please add suggestions for the user to it. 
1. Do we do not have any heterozygous variants in the VCF file?
2. Is the VCF file not correctly formated?
3. Do we have issues with the chromosome style? (Not sure if this is really a problem in this script)
4.  ....

**Review by c-mertes — CHANGES_REQUESTED** [review]
The commits are fine just expend the error message for the user.

I just have one issue to raise here, but do not have the best solution yet. If you throw an error, the user can not finish the pipeline. Hence, he will not get the nice HTML reports before fixing those issues nor get a result file as it depends on all input of all samples. If you are happy with this for the end-user go ahead. 

Maybe @vyepez88 has more throughs on this.

File: drop/modules/mae-pipeline/MAE/filterSNVs.sh (hunk 2 above)
Comment (vyepez88) [reply drop/modules/mae-pipeline/MAE/filterSNVs.sh]: number 3. shouldn't be a problem in this script.
4. Check if vcf file exists and its index exists (although the pipeline might not even begin if this is not the case)

File: drop/modules/mae-pipeline/MAE/ASEReadCounter.sh (hunk 1 above)
Comment (vyepez88) [reply drop/modules/mae-pipeline/MAE/ASEReadCounter.sh]: 1. This shouldn't happen with the new fixes
2. Nick is implementing this feature

**Comment by vyepez88** [comment]
I'm fine with the users not getting the aggregated results table + html. They can anyway get the results of all the other samples by running the pipeline with the `-k` command. But in my experience, it either works for all files, or doesn't work at all.

**Comment by mumichae** [comment]
@c-mertes Would you suggest to change this error message to a warning then and deal with empty ouputs later in the pipeline?

**Review by c-mertes — APPROVED** [review]
looks all good now. thanks @mumichae

**Pull request merged**